lib/libc/riscv/sys/cerror.S

```diff
@@ -34,22 +34,22 @@
 
 #include <machine/asm.h>
 __FBSDID("$FreeBSD$");
+#include "SYS.h"
+
+#define FRAME_SIZE (2 * __SIZEOF_POINTER__)
+#define RA_OFFSET 0
+#define ERRNO_OFFSET __SIZEOF_POINTER__
 
 ENTRY(cerror)
-#ifdef __CHERI_PURE_CAPABILITY__
-#pragma message("TODO: RISCV-PURECAP implement cerror")
-	cret
-#else
-	addi	sp, sp, -16
-	sd	a0, 0(sp)
-	sd	ra, 8(sp)
-	call	_C_LABEL(__error)
-	ld	a1, 0(sp)
-	ld	ra, 8(sp)
-	sw	a1, 0(a0)
-	li	a0, -1
-	li	a1, -1
-	addi	sp, sp, 16
-	ret
-#endif
+	ADJUST_SP(-FRAME_SIZE)
+	STORE_PTR(ra, RA_OFFSET, sp)
+	STORE_32(a0, ERRNO_OFFSET, sp)	# save errno value
```

Comment:
Unrelated to this patch, why is this spilling rather than using a saved register?

Reply:
We would still have to save the callee-save register?

Reply:
Oh, that's true, so might as well let the callee do it.